rim/static/rim/js/base.js

```diff
@@ -20,4 +20,65 @@ $(document).ready(function() {
         }
     })
 
+    var timer;
+    $('#id_serial_no').on('input', function(e) {
+        clearTimeout(timer)
+        timer = setTimeout(check_serial_nums, 300);
+    })
+
+    function check_serial_nums() {
+        var existing_serial_nums = [];
+        $('#serial_no_errors').remove()
+        $.ajaxSetup({
+            headers: { "X-CSRFToken": getCookie("csrftoken") }
+        });
+        $.ajax({
+            url: "/check_serial_nums/",
+            data: {'serial_nums': JSON.stringify($('#id_serial_no').val().trim().split('\n'))},
+            type: 'POST',
+            success: function(data) {
+                existing_serial_nums = data['context'];
+                if (existing_serial_nums.length != 0) {
+                    add_errors(existing_serial_nums);
+                }
+            }
+        });
+    }
+
+    function getCookie(c_name) {
+        if (document.cookie.length > 0)
+        {
+            c_start = document.cookie.indexOf(c_name + "=");
+            if (c_start != -1)
+            {
+                c_start = c_start + c_name.length + 1;
+                c_end = document.cookie.indexOf(";", c_start);
+                if (c_end == -1) c_end = document.cookie.length;
+                return unescape(document.cookie.substring(c_start,c_end));
+            }
+        }
+        return "";
+    }
```

Comment:
This function, along with the `$.ajaxSetup()` code from `check_serial_nums()` can stay in base.js. Although you might want to update it with the [latest version from the Django docs](https://docs.djangoproject.com/en/3.0/ref/csrf/#ajax):

```suggestion
    function getCookie(name) {
        var cookieValue = null;
        if (document.cookie && document.cookie !== '') {
            var cookies = document.cookie.split(';');
            for (var i = 0; i < cookies.length; i++) {
                var cookie = cookies[i].trim();
                // Does this cookie string begin with the name we want?
                if (cookie.substring(0, name.length + 1) === (name + '=')) {
                    cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
                    break;
                }
            }
        }
        return cookieValue;
    }
    var csrftoken = getCookie('csrftoken');
        function csrfSafeMethod(method) {
        // these HTTP methods do not require CSRF protection
        return (/^(GET|HEAD|OPTIONS|TRACE)$/.test(method));
    }
    $.ajaxSetup({
        beforeSend: function(xhr, settings) {
            if (!csrfSafeMethod(settings.type) && !this.crossDomain) {
                xhr.setRequestHeader("X-CSRFToken", csrftoken);
            }
        }
    });
```

Leaving it in the base.js should automatically add the CSRF header for any future AJAX code we happen to write.